src/Perl6/Metamodel/EnumHOW.nqp

```diff
@@ -85,11 +96,17 @@ class Perl6::Metamodel::EnumHOW
     }
 
     method enum_from_value($obj, $value) {
-        unless %!value_to_enum {
-            for @!enum_value_list {
-                %!value_to_enum{$_.value} := $_;
+        my %value_to_enum := %!value_to_enum;
+        $lock.protect: {
+            unless %value_to_enum {
+                for @!enum_value_list {
+                    %value_to_enum{$_.value} := $_;
+                }
+                nqp::scwbdisable();
+                %!value_to_enum := %value_to_enum;
+                nqp::scwbenable();
             }
-        }
+        };
```

Comment:
I think a better fix here would be to:

1. Switch to a `$` attribute
2. Create the hash in a lexical
3. Bind the hash into the attribute

Of course two threads may produce it in duplicate, but that's not a correctness issue, and it's almost certainly still more efficient than having a Global Enum Lock.